pkg/operator/staticpod/controller/installer/installer_controller_test.go

```diff
@@ -884,6 +885,37 @@ func TestCreateInstallerPodMultiNode(t *testing.T) {
 			updateStatusErrors:   []error{errors.NewInternalError(fmt.Errorf("unknown"))},
 			expectedSyncError:    []bool{true},
 		},
+		{
+			name:                    "three nodes, 2 not updated, one with failure in last revision which is no longer available",
+			latestAvailableRevision: 3,
+			nodeStatuses: []operatorv1.NodeStatus{
+				{
+					NodeName:        "test-node-0",
+					CurrentRevision: 1,
+				},
+				{
+					NodeName:           "test-node-1",
+					CurrentRevision:    1,
+					LastFailedRevision: 2,
```

Comment:
I thought the case we look at has target version 2, and the owner ref code fails, but we still reach revision 3 eventually.

Reply:
Yes, that is what is happening in this test.

Reply:
Updated.

Reply:
LastFailedRevision=2 and TargetRevision=2 at the same time is not a valid state. Only set TargetRevision.

Reply:
@sttts Updated.

Reply:
@sttts Updated.